test/extended/util/test.go

```diff
@@ -453,6 +453,21 @@ var (
 // checkSyntheticInput selects tests based on synthetic skips or focuses
 func checkSyntheticInput() {
 	checkSuiteSkips()
+	checkSkipTestsEnvVar()
```

Comment:
So it looks like `checkSuiteSkips()` is designed to figure out certain tests to skip where it can't figure that out based on the test name. For name-based stuff, I think it would be better to tie into `AnnotateTestSuite()`, and just label the relevant tests with a `[Disabled:Something]` label at the start, rather than figuring out that the test needs to be skipped just-in-time like `checkSuiteSkips()` is doing (and having to reparse the environment variable again for every test).

Reply:
hmm, yeah, makes sense, seems better spot.